server/service/TcpClientService.js

```diff
@@ -115,52 +120,66 @@ exports.putTcpClientRemoteAddress = function(url,body,uuid) {
  * uuid String 
  * no response value expected for this operation
  **/
-exports.putTcpClientRemotePort = function (url, body,uuid) {
+exports.putTcpClientRemotePort = function (url, body, uuid) {
   return new Promise(async function (resolve, reject) {
     try {
-      let isUpdated = await fileOperation.writeToDatabaseAsync(url, body, false);
-
- 
-
-      /****************************************************************************************
-       * Prepare attributes to automate forwarding-construct
-       ****************************************************************************************/
-      if(isUpdated){
-        let forwardingAutomationInputList = await prepareForwardingAutomation.OAMLayerRequest(
-          uuid
-        );
-        ForwardingAutomationService.automateForwardingConstructWithoutInputAsync(
-          forwardingAutomationInputList
-        );
-      }      
+      let oldValue = await tcpClientInterface.getRemotePortAsync(uuid);
+      let newValue = body["tcp-client-interface-1-0:remote-port"];
+      if (oldValue !== newValue) {
+        let isUpdated = await fileOperation.writeToDatabaseAsync(url, body, false);
+
+        /****************************************************************************************
+         * Prepare attributes to automate forwarding-construct
+         ****************************************************************************************/
+        if (isUpdated) {
+          let forwardingAutomationInputList = await prepareForwardingAutomation.OAMLayerRequest(
+            uuid
+          );
+          ForwardingAutomationService.automateForwardingConstructWithoutInputAsync(
+            forwardingAutomationInputList
+          );
+        }
+      }
       resolve();
-    } catch (error) {}
+    } catch (error) { }
     reject();
   });
 }
 
-exports.putTcpClientRemoteProtocol = function(url,body,uuid) {
-  return new Promise(async function(resolve, reject) {
-  
-      try {
+exports.putTcpClientRemoteProtocol = function (url, body, uuid) {
+  return new Promise(async function (resolve, reject) {
+
+    try {
+      let oldValue = await tcpClientInterface.getRemoteProtocolAsync(uuid)
+      let newValue = body["tcp-client-interface-1-0:remote-protocol"];
+      let value
+      if (oldValue == "HTTPS") {
+        value = "tcp-client-interface-1-0:PROTOCOL_TYPE_HTTPS"
```

Comment:
can we get the value from the enum declared in tcpClientInterface instead of hardcoding here